pytest_terraform/tf.py

```diff
@@ -386,11 +386,15 @@ def create(self, request, module_dir):
         if self.teardown_config != td.OFF:
             request.addfinalizer(self.tear_down)
         try:
-            test_api = self.runner.apply()
-            tfstatejson = test_api.save()
+            terraform_state = self.runner.apply()
+            tfstatejson = terraform_state.save()
+            test_api = TerraformTestApi.from_string(tfstatejson)
+
             self.config.hook.pytest_terraform_modify_state(tfstate=tfstatejson)
-            test_api.update(tfstatejson)
-            test_api.save(module_dir.join("tf_resources.json"))
+
+            terraform_state.update(tfstatejson)
```

Comment:
this api pattern feels way to subtle.. ideally we'd have a serialization request thats gone through hooks that we directly save out, and not update/modify terraform_state at all with the modifications.

Reply:
at the start of this code block we take the terraform_state from the runner and create TerraformStateJson object, that then gets fed to a new instance of TerraformStateApi as test_api which takes a copy of that object to return to the function under test and the TerraformState object gets updated with any mutation from the hook and finally saved.